public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -98,6 +114,7 @@ export const GeneratePopoverBody: React.FC<{
               `Please provide your insight on this ${summaryType}.`
             );
           }
+          reportGeneratedMetric('generated');
```

Comment:
nit: should we have seprate metric for alert summary and insight?

Reply:
Should we report this metric if error happened? And what if summary success while insight fails?

If we need to report metrics no matter it succeed or not, we should move this logic to `finally` block.

Reply:
I think we should only report when the summary is generated successfully.